sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -545,7 +545,11 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
   }
 
   private def getLocationFromStorageProps(table: CatalogTable): Option[String] = {
-    CaseInsensitiveMap(table.storage.properties).get("path")
+    if (conf.get(HiveUtils.FOLLOW_TABLE_LOCATION)) {
```

Comment:
It looks tricky to have a config for such a subtle behavior. I think it makes more sense to fail when the path serde property doesn't match the table location, and ask users to either correct the path property or the table location.

Reply:
@cloud-fan Thank you for your suggestion, I have corrected.